content/posts/numpy/numpy2/index.md

```diff
@@ -0,0 +1,113 @@
+  [NEP 55](https://numpy.org/neps/nep-0055-string_dtype.html). And it is our hope
+  that enable future new data types with interesting new capabilities in the
+  PyData ecosystem and in NumPy itself.
+
+- Windows compatibility enhancements: The default 32-bit integer representation
+  on Windows has been updated to 64-bit on 64-bit architectures, addressing one
+  of the most common issues a developer runs into trying to get NumPy code to
+  work portably on Windows and Unix-like operating systems.
+
+- Full support for the Python array API standard: This is the first release to
+  include full support for the array API standard (v2022.12), which was enabled
+  by the promotion rules and API cleanup mentioned above, as well as by
+  adding new APIs and aligning existing APIs and behavior with the standard,
+  as proposed by [NEP 56](https://numpy.org/neps/nep-0056-array-api-main-namespace.html).
+
+These are just some of the more impactful changes in behavior and usability. In addition,
+NumPy 2.0 contains significant performance improvements, large documentation improvements,
+and more much - for a more extensive list of changes, see
+the [NumPy 2 release notes](https://numpy.org/devdocs/release/2.0.0-notes.html).
+
+A major release comes with changes that users may have to adapt to, but we
+worked hard to strike a balance between improvements and ensuring that the
+transition to NumPy 2.0 is as seamless as possible. We wrote a comprehensive
+[migration guide](https://numpy.org/devdocs/numpy_2_0_migration_guide.html),
+and a [ruff plugin](https://numpy.org/devdocs/numpy_2_0_migration_guide.html#ruff-plugin)
+helps to automatically update Python code so it will work with both NumPy 1.x and
+NumPy 2.x.
+
+We knew when we started the development for 2.0 that rolling out a NumPy 2.0
+will be (temporarily) disruptive, because of the backwards-incompatible ABI change
+in particular which prevents previously-released wheels of other packages
+that use the NumPy C API from working with numpy 2.0 wheels in the PyPI ecosystem.
+We spent an extraordinary amount of effort communicating these changes, helping
+downstream projects adapt, tracking compatibility of popular open source projects
+(see, e.g., [numpy#26191](https://github.com/numpy/numpy/issues/26191), and
+completing the release process at a very mild pace to give everyone enough time
+to adapt. No doubt the next few weeks will be slightly rocky still, however
+we expect this to be manageable and well worth it in the long run.
+
+While we do require C API users to recompile their projects to support running
+with NumPy 2.0, we prepared for this already in NumPy 1.25. The build process was
+simplified so that you can now always compile with the latest NumPy version.
+This means that projects build with NumPy 2.x are "magically" compatible with
+1.x. It also means that projects no longer need to build their binaries using
+the oldest version of NumPy supported by a project.
```

Comment:
It's OK, but it feels now a bit much to put this after the other paragraph. 

Maybe swapping the two paragraphs makes sense (use the C "incompatible ABI phrasing from the first there).

Then follow with the "extraordinary amount of effort communicating these changes, which doesn't really need to be C API specific.  (There was also a lot of Python effort, it just stretched a longer time period.)

Reply:
Sounds good. I wasn't quite sure about the order, will swap.

Reply:
done now, please have another look

Reply:
Looks fine, I was hoping to also shorten it a bit, but no great idea right now.

Reply:
I guess it is a bit shorter :).

Reply:
Yep, a little:)

Reply:
same here, can you resolve @seberg?